src/lucky_record/errors.cr

```diff
@@ -5,11 +5,15 @@ module LuckyRecord
   class LuckyRecordError < Exception
   end
 
-  # Raised when Lucky Record cannot find a record by given id
+  # Raised when a record could not be find
   class RecordNotFoundError < LuckyRecordError
     def initialize(@model : Symbol, @id : String)
       super "Could not find #{model} with id of #{id}"
     end
+
+    def initialize(@model : Symbol, @id : Symbol)
```

Comment:
Instead of calling it id, what do you think about “query”? Not perfect but maybe a bit more clear than id?

Reply:
Yeah it's a bit clearer.

Reply:
Okay, so if I rename the overload to `query` then I need to make the first `id` argument nilable.

```bash
The instance variable '@id' is initialized in other 'initialize' methods,
and by not initializing it here it's not clear if the variable is supposed
to be nilable or if this is a mistake.

To fix this error, either assign nil to it here:

  @id = nil

Or declare it as nilable outside at the type level:

  @id : (String)?

    def initialize(@model : Symbol, @query : Symbol)

```

Reply:
That's a good error and it highlights that we probably don't need instance variables. You can use local variables instead and the issue should go away

Reply:
Okay so its because the instance variable is set on the class and needs to be typed, got it

Reply:
Kind of. The problem was that `@id` was not nilable, but if you called the `def initialize(@model : Symbol, @query : Symbol)` then `@id` would be `nil`. So you it was saying you either needed to set the variable or make it nilable with String?

By using local vars there is no issue because not all initializers need to set them